src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,62 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String COMMA_SEPARATOR = ",";
+    private static final int SUPPLY_INDEX = 0;
+    private static final int BUY_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeToFile(
+                composeReport(
+                readFromFile(fromFileName)), toFileName);
+    }
+
+    public void writeToFile(String data, String toFileName) {
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFileName))) {
+            bufferedWriter.write(data);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write data to " + toFileName, e);
+        }
+    }
+
+    public String readFromFile(String fromFileName) {
+        String result;
+        try {
+            result = Files.readString(Path.of(fromFileName))
+                    .replaceAll("\\r?\\n|\\r", COMMA_SEPARATOR);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file " + fromFileName, e);
+        }
+        return result;
+    }
+
+    public String composeReport(String data) {
+        String [] tmp = data.split(COMMA_SEPARATOR);
+        int [] report = new int[2];
```

Comment:
Pay more attention to the names. int array **is not** a `report`. `String [] tmp` - unclear name.
```suggestion
        int supplySum = 0;
        int buySum = 0;
```